daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -95,6 +95,52 @@ func init() {
 	}
 }
 
+// eventBatch holds the events that are batched for submission and the
+// associated data about it.  This type is not threadsafe and must not
+// be used concurrently.
+type eventBatch struct {
+	events []wrappedEvent
+	bytes  int
+}
+
+func newEventBatch() *eventBatch {
+	return &eventBatch{
+		events: make([]wrappedEvent, 0),
+		bytes:  0,
+	}
+}
+
+// Events returns the sorted slice of wrappedEvents.
+func (b *eventBatch) Events() []wrappedEvent {
+	// events in a batch must be sorted by timestamp
+	// see http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
+	sort.Sort(byTimestamp(b.events))
```

Comment:
Isn't this undoing https://github.com/moby/moby/commit/443f251cf596006fe4cb621dcab955f7da0a2a00 (https://github.com/moby/moby/pull/24814)? /cc @samuelkarp

Reply:
This is respecting that, the code that stable sorts according to timestamp and then insertion order is implemented in the sort for this type (https://github.com/jahkeup/moby/blob/b0046c3e68579640b5b08a4cb8b754924fbc2a30/daemon/logger/awslogs/cloudwatchlogs.go#L631).

The insertion order is recorded as the wrappedEvent is crafted.

Reply:
Ah, thanks! Sorry, it wasn't clear directly 😊 Do you think we need to change `byTimestamp` to something else? Alternatively, move the description of sorting to the GoDoc of the `Events()` method, and describe that it's sorting by timestamp, but preserving `insertOrder` if timestamp is equal

Reply:
I'll update the GoDoc for this method to clarify the action and results of retrieving the events from the batch.

The `byTimestamp` type could probably be better named for its usages, however I can't think of a more apt name without stringing together many many words. Do you have any suggestions on a different name? Do you think the GoDoc statement could instead be more clear without needing a name change?